lib/util.js

```diff
@@ -45,6 +45,10 @@ module.exports.RestError = class RestError extends Error {
             super(ex.message);
             this.code = ex.code;
         }
+        if (url) {
+            this.endpoint = url;
+            this.message += ` (endpoint: ${this.endpoint})`;
```

Comment:
while endpoint makes sense, not sure customizing message does, because it means it is not the message returned. this should be handled in any app using the lib

Reply:
```suggestion
```